openwisp_controller/notifications/apps.py

```diff
@@ -0,0 +1,68 @@
+from django.apps import AppConfig
+from django.utils.translation import gettext_lazy as _
+from openwisp_notifications.types import (
+    register_notification_type,
+    unregister_notification_type,
+)
+
+
+class NotificationsConfig(AppConfig):
+    name = 'openwisp_controller.notifications'
+
+    def ready(self):
+        from . import handlers  # noqa
+
+        self.register_notification_types()
+
+    def register_notification_types(self):
+        unregister_notification_type('default')
+        register_notification_type(
+            'config_problem',
```

Comment:
config_error

Reply:
I was trying to be consistent with notification types in openwisp-monitoring.

https://github.com/openwisp/openwisp-monitoring/blob/d022aedba3aa4bc8cad5be3f9bb1edb69481f247/openwisp_monitoring/monitoring/configuration.py#L377-L385

Isn't introducing these notification types here an overlapping effort?:
https://github.com/openwisp/openwisp-monitoring/blob/d022aedba3aa4bc8cad5be3f9bb1edb69481f247/openwisp_monitoring/monitoring/configuration.py#L117-L126

Is it because, we don't want users to rely on openwisp-monitoring for notifications? Please correct me if I am wrong.  I will test this branch in openwisp-monitoring and see if sends two notifications for every such event. 

@nepython maybe you can help me here in understanding when such notifications will be created in openwisp-monitoring. :sweat_smile:

Reply:
@pandafy, config status can be:
`applied`: If everything is perfect
`modified`: This status is shown when config has been modified but not applied
`error`: When any problem popped up while attempting to apply the `Config`
Right now, if config status changes from `applied` to `modified` or `error` then notification of `config_applied_problem` type which you have correctly linked would be sent out by the monitoring module. So *I think* after this PR the part of sending notification to the user after status turning to `error` would be handled by `openwisp-controller` and only `problem` status would be handled by monitoring module (reason being decoupling the same from monitoring module https://github.com/openwisp/openwisp-monitoring/issues/195).

This is how I view it, @nemesisdesign might be able to confirm / correct me.

Reply:
> So _I think_ after this PR the part of sending notification to the user after status turning to `error` would be handled by `openwisp-controller` and only `problem` status would be handled by monitoring module

OpenWISP Monioring performs an active check periodically to find out if a config has stayed in "modified" status for more than 5 minutes, which indicates [openwisp-config](https://github.com/openwisp/openwisp-config/) is not working as intended.

OpenWISP Controller should just notify any time the config moves into error status.

Reply:
Thank you Federico and Hardik for clearing my doubts.